src/ui/components/search/searchcomponent.js

```diff
@@ -348,26 +376,46 @@ export default class SearchComponent extends Component {
     if (this.iconState === iconState) {
       return;
     }
+    this.prevState = this.iconState;
     this.iconState = iconState;
     if (!this.isRequestingAnimationFrame) {
       this.isRequestingAnimationFrame = true;
       window.requestAnimationFrame(() => {
-        this.forwardIcon.classList.remove('yxt-SearchBar-AnimatedIcon--paused');
-        this.reverseIcon.classList.remove('yxt-SearchBar-AnimatedIcon--paused');
+        DOM.query(this._container, this.prevState).classList.add('yxt-SearchBar-Icon--inactive');
+        const iconEl = DOM.query(this._container, this.iconState);
+        iconEl.classList.remove('yxt-SearchBar-Icon--inactive');
+        iconEl.classList.remove('yxt-SearchBar-AnimatedIcon--paused');
```

Comment:
would it be possible to remove `yxt-SearchBar-AnimatedIcon--paused` only when the iconState is MAGNIFYING_GLASS or YEXT?